torch/cuda/__init__.py

```diff
@@ -1137,6 +1137,46 @@ def _dtype(self):
 torch._storage_classes.add(ComplexDoubleStorage)
 torch._storage_classes.add(ComplexFloatStorage)
 
+
+class _WrappedTritonKernel(object):
```

Comment:
nit: we should move all of this to triton specific code.
It makes sense to me to have this file check the triton backends file

Reply:
Actually the name is wrong, it's just a `CallableInvokedWrapper`, isn't it? (I..e any callable can be decorate with it  to allow one check whether callable was invoked or not)

Reply:
Actually, this has nothing to do with Triton, i.e. it should be called something like `IsCallableInvokedWrapper`, because it can be attached to any callable to inspect whether or not it was called by anyone.

Reply:
If we have a generic util file we can put that there for sure. If we only use it within these triton bindings, it's fine to put it there for now I think.

Reply:
@albanD @malfet - How should @nikitaved best address this?